### ISSUE TITLE
`jsdoc/check-template-names` reports used template as unused

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [gajus/eslint-plugin-jsdoc#1286]
## Expected behavior
No warnings. Template `T` is in use and the function type checks with `// @ts-check`. I would expect both `@param`s and maybe the `@returns` to separately be enough to mark the template as used.

The `@type` inside the function also references it, but that probably shouldn't be considered as the template should already be part of the function signature.

## Actual behavior
Starting from `eslint-plugin-jsdoc` version `48.10.0`, `jsdoc/check-template-names` reports that `@template T not in use`.

## ESLint Config

```json
{
	"root": true,
	"parserOptions": {
		"ecmaVersion": "latest",
		"sourceType": "module"
	},
	"plugins": ["jsdoc"],
	"settings": {
		"jsdoc": {
			"mode": "typescript"
		}
	},
	"env": { "es2024": true },
	"rules": {
		"jsdoc/check-template-names": "warn"
	}
}
```

## ESLint sample

```js
/**
 * Uses the provided callback to group the given array into the keys of a map.
 * Based on the array grouping proposal: https://github.com/tc39/proposal-array-grouping/
 *
 * @template T
 * @param {T[]} array
 * @param {(value: T, index: number) => string} callbackFn
 * @returns {Map<string, T[]>}
 */
export function mapGroupBy(array, callbackFn) {
	/** @type {Map<string, T[]>} */
	const map = new Map();
	for (const [index, element] of array.entries()) {
		const key = callbackFn(element, index);
		const arr = map.get(key);
		if (arr) {
			arr.push(element);
		} else {
			map.set(key, [element]);
		}
	}
	return map;
}
```

## Environment

- Node version: `v20.16.0`
- ESLint version `v8.57.0`
- `eslint-plugin-jsdoc` version: `48.10.1`

**Comment by ehoogeveen-medweb** [comment]
> The `@type` inside the function also references it, but that probably shouldn't be considered as the template should already be part of the function signature.

Having said that, without the `@returns` annotation the `@type` does specify the type for the return value, which could be the only usage of `T` (and would be valid TypeScript).

**Comment by brettz9** [comment]
Thanks for the report!

**Comment by ehoogeveen-medweb** [comment]
Thanks for the quick fix!
